cleverhans/attacks_tf.py

```diff
@@ -57,6 +63,38 @@ def fgsm(x, predictions, eps, clip_min=None, clip_max=None):
     return adv_x
 
 
+def fgsm_iterative(x, y, model, eps, eps_iter, n_iter,
+                   clip_min=None, clip_max=None):
+    """
+    TensorFlow implementation of the iterative Fast Gradient
+    Sign method.
+    :param x: the input placeholder
+    :param y: the output placeholder
+    :param model: the Keras model that we want to attack
+    :param eps: the maximum input variation parameter
+    :param eps_iter: the input variation for each iteration
+    :param n_iter: the number of iterations to run
+    :param clip_min: optional parameter that can be used to set a minimum
+                    value for components of the example returned
+    :param clip_max: optional parameter that can be used to set a maximum
+                    value for components of the example returned
+    :return: a tensor for the adversarial example
+    """
+    # define the upper and lower bounds
+    upper_bound = x + eps
+    lower_bound = x - eps
+    # run through the iterations
+    adv_x = x
+    for i in range(n_iter):
+        predictions = model(adv_x)
+        adv_x = fgsm(adv_x, predictions, eps_iter,
+                     clip_min=clip_min, clip_max=clip_max, y=y)
+        adv_x = tf.minimum(adv_x, upper_bound)
```

Comment:
I'm not sure whether it can be done with `tf.clip`. We need a different value at every feature. `tf.clip_by_value` takes a single value for the whole vector. So this can't be folded into `clip_min` and `clip_max` either since they use `tf.clip_by_value`